applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_operations.cpp

```diff
@@ -0,0 +1,166 @@
+    
+    // Variables addition
+    model_part.AddNodalSolutionStepVariable(VELOCITY_POTENTIAL);
+    model_part.AddNodalSolutionStepVariable(AUXILIARY_VELOCITY_POTENTIAL);
+    
+    // Set the element properties
+    model_part.CreateNewProperties(0);
+    Properties::Pointer pElemProp = model_part.pGetProperties(0);
+   
+    // Geometry creation
+    model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+    model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+    model_part.CreateNewNode(3, 1.0, 1.0, 0.0);
+    std::vector<ModelPart::IndexType> elemNodes_1{ 1, 2, 3 };
+    model_part.CreateNewElement("IncompressiblePotentialFlowElement2D3N", 1, elemNodes_1, pElemProp);
+    auto& r_element = model_part.GetElement(1);
+    r_element.GetValue(WAKE) = 1;
+    Vector wake_elemental_distances(3);
+    wake_elemental_distances(0) = 1.0;
+    wake_elemental_distances(1) = 1.0;
+    wake_elemental_distances(2) = -1.0;
+    r_element.GetValue(WAKE_ELEMENTAL_DISTANCES) = wake_elemental_distances;
+    
+    // Define the nodal values
+    Vector potential(3);
+    potential(0) = 1.0;
+    potential(1) = 2.0;
+    potential(2) = 3.0;
+    for (unsigned int i = 0; i < 3; i++){
+      if (wake_elemental_distances(i) > 0.0)
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential(i);
+      else
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL) = potential(i);
+    }
+    for (unsigned int i = 0; i < 3; i++){
+      if (wake_elemental_distances(i) < 0.0)
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = potential(i)+1;
+      else
+        r_element.GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL) = potential(i)+1;
+    }      
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    
+    // Create fake compressible_model_part
+    ModelPart& compressible_model_part = this_model.CreateModelPart("Compressible", 3);
+    compressible_model_part.GetProcessInfo()[DOMAIN_SIZE] = 2;
+    
+    // Variables addition
+    compressible_model_part.AddNodalSolutionStepVariable(DENSITY);
+    compressible_model_part.AddNodalSolutionStepVariable(MOMENTUM);
+    compressible_model_part.AddNodalSolutionStepVariable(TOTAL_ENERGY);
+    
+    // Set the element properties
+    compressible_model_part.CreateNewProperties(0);
+    Properties::Pointer compressible_pElemProp = compressible_model_part.pGetProperties(0);
+    
+    // Geometry creation
+    compressible_model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+    compressible_model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+    compressible_model_part.CreateNewNode(3, 1.0, 1.0, 0.0);
+    std::vector<ModelPart::IndexType> elemNodes_2{ 1, 2, 3 };
+    compressible_model_part.CreateNewElement("IncompressiblePotentialFlowElement2D3N", 1, elemNodes_2, compressible_pElemProp);
```

Comment:
```suggestion
    compressible_model_part.CreateNewElement("Element2D3N", 1, elemNodes_2, compressible_pElemProp);
```
It makes no difference at all for the sake of testing, but to make it more clear I'd use the fake element implementation in this case.